api/v1alpha1/argocd_types.go

```diff
@@ -115,6 +115,18 @@ type ArgoCDApplicationControllerShardSpec struct {
 
 	// Replicas defines the number of replicas to run in the Application controller shard.
 	Replicas int32 `json:"replicas,omitempty"`
+
+	// EnableDynamicScaling defines whether dynamic scaling should be enabled for Application Controller component
+	EnableDynamicScaling bool `json:"enableDynamicScaling,omitempty"`
```

Comment:
Should this be renamed as `DynamicScalingEnabled` ? I see an existing property `StatusBadgeEnabled` and thought we can follow a similar naming convention.